src/main.ts

```diff
@@ -0,0 +1,137 @@
+import {NotificationMessage} from './NotificationMessage';
+import {EventType} from "./enums";
+import {schemaOnIssue, schemaOnIssueComment, schemaOnPullRequest, schemaOnPush} from "./eventPayloadSchemaBuilder";
+
+const core = require ('@actions/core');
+const path = require('path');
+const https = require('https');
+
+const jsonPath = path.join(__dirname, '..', 'resources', 'notification.json');
+const workflow = process.env.GITHUB_WORKFLOW;
+const repository = process.env.GITHUB_REPOSITORY;
+const branch = process.env.GITHUB_REF;
+const github_event_payload =  process.env.GITHUB_EVENT_PATH || jsonPath;
+const trigger_event_name = process.env.GITHUB_EVENT_NAME;
+
+const msteams_webhook_url = core.getInput('webhook_url');
+const job_status = core.getInput('job_status');
+const event_id = core.getInput('event_id');
+
+export const parsedNotificationMessage = async function parse_event_to_message(event_payload_text: string): Promise<NotificationMessage> {
+    let details;
+    let account;
+    let message;
+    let url;
+    let parsedSchema;
+    try {
+        if (job_status.toUpperCase() === "CANCELLED") {
+            console.log("Job was cancelled, no notification will be sent")
+            process.exit(0);
+        }
+
+        let event_name : string = trigger_event_name || "event";
+        if (event_id) {
+            event_name = event_id;
+        }
+
+        switch(event_name){
+            case EventType.PUSH:
+                await schemaOnPush.validate(event_payload_text);
```

Comment:
why is schema validation async? Seems like an unnecessary complication

Reply:
Man, you are quick! I was actually thinking about that, there is a validateSync option so I am giving it a try as well. I thought of the I/O operation we need to perform to read the file from the file system (the payload is saved in a local file on the VM) --> obviously I would prefer the synchronous operations and no more promises, please!

Reply:
I ran some tests between _validateSync_ and _validate_ and to be honest I do not see a difference in the result - only in the implementation of course because we skip Promises. Therefore I opted for validateSync. I have my doubts on whether we are really "promise"- and "async"-free, as the official yup documentation states : 
_Runs validatations synchronously **if possible** and returns the resulting value, or throws a ValidationError. Accepts all the same options as validate._
(I am referring to the if possible part).